.github/workflows/check-publish.yml

```
@@ -0,0 +1,28 @@
+name: Check publish
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  check-publish:
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
+    runs-on: ${{ matrix.os }}
+    container:
+      image: paritytech/ci-unified:bullseye-1.70.0-2023-05-23-v20230706
+    steps:
+      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@a95ba195448af2da9b00fb742d14ffaaf3c21f43
```

```suggestion
        uses: Swatinem/rust-cache@3cf7f8cc28d1b4e7d01e3783be10a97d55d483c8 # v2.7.1
```

While PR being reviewed newer version of `Swatinem/rust-cache` came out. It fixes `Cargo.toml` parsing issues. This PR also [affected](https://github.com/paritytech/polkadot-sdk/actions/runs/6640336626/job/18040548683?pr=1887#:~:text=9-,Warning%3A%20Error%20parsing%20Cargo.toml%20manifest%2C%20fallback%20to%20caching%20entire%20file%3A%20Error%3A%20Invalid%20TOML%20document%3A%20expected%20key%2Dvalue%2C%20found%20comma,-10)